modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/CompositeArtifactExtensionManager.java

```diff
@@ -58,8 +58,8 @@ public CompositeArtifactExtensionManager(ExtensionManager parentExtensionManager
     this.childExtensionManager = childExtensionManager;
 
     extensionModels = new HashSet<>();
-    extensionModels.addAll(parentExtensionManager.getExtensions());
     extensionModels.addAll(childExtensionManager.getExtensions());
+    extensionModels.addAll(parentExtensionManager.getExtensions());
```

Comment:
why this change? this looks extra suspicious considering you're adding into a HashSet

Reply:
Classloaders for each extension are set using a property of their extension models. This change aims to use the policy classloader when the policy is executing and not the application one (note that the set uses the extension name as the key, so the order matters, because in the case of an application declaring the same extension than a policy, the policy extension model was not added to this set).

Reply:
This is not the correct way of achieving what you want:

* This is a HashSet, order is not guaranteed. This is would actually be flaky on different depending on the JDK version/vendor to use
* If you want a specific classloader to be used, then you need to reference it specifically (usually by setting it as the TCCL).

Reply:
After a refactor in order to address the backward compatibility issue, this discussion no longer applies.